tests/aws/stepfunctions/v2/test_sfn_api_versioning.py

```diff
@@ -0,0 +1,577 @@
+            stateMachineArn=state_machine_arn
+        )
+        sfn_snapshot.match("publish_v1", publish_v1)
+        state_machine_v1_arn = publish_v1["stateMachineVersionArn"]
+
+        await_state_machine_version_listed(
+            aws_client.stepfunctions, state_machine_arn, state_machine_v1_arn
+        )
+
+        list_versions_resp_1 = aws_client.stepfunctions.list_state_machine_versions(
+            stateMachineArn=state_machine_arn
+        )
+        sfn_snapshot.match("list_versions_resp_1", list_versions_resp_1)
+
+        describe_v1 = aws_client.stepfunctions.describe_state_machine(
+            stateMachineArn=state_machine_v1_arn
+        )
+        sfn_snapshot.match("describe_v1", describe_v1)
+
+        definition_r2 = BaseTemplate.load_sfn_template(BaseTemplate.BASE_PASS_RESULT)
+        definition_r2["Comment"] = f"{definition_r2['Comment']}-R2"
+        definition_r2_str = json.dumps(definition_r2)
+
+        update_resp_2 = aws_client.stepfunctions.update_state_machine(
+            stateMachineArn=state_machine_arn, definition=definition_r2_str
+        )
+        sfn_snapshot.match("update_resp_2", update_resp_2)
+        revision_id_r2 = update_resp_2["revisionId"]
+
+        publish_v2 = aws_client.stepfunctions.publish_state_machine_version(
+            stateMachineArn=state_machine_arn, description="PublishedV2Description"
+        )
+        sfn_snapshot.match("publish_v2", publish_v2)
+        state_machine_v2_arn = publish_v2["stateMachineVersionArn"]
+
+        await_state_machine_version_listed(
+            aws_client.stepfunctions, state_machine_arn, state_machine_v2_arn
+        )
+
+        list_versions_resp_2 = aws_client.stepfunctions.list_state_machine_versions(
+            stateMachineArn=state_machine_arn
+        )
+        sfn_snapshot.match("list_versions_resp_2", list_versions_resp_2)
+
+        describe_v2 = aws_client.stepfunctions.describe_state_machine(
+            stateMachineArn=state_machine_v2_arn
+        )
+        sfn_snapshot.match("describe_v2", describe_v2)
+
+        definition_r3 = BaseTemplate.load_sfn_template(BaseTemplate.BASE_PASS_RESULT)
+        definition_r3["Comment"] = f"{definition_r3['Comment']}-R3"
+        definition_r3_str = json.dumps(definition_r3)
+
+        update_resp_3 = aws_client.stepfunctions.update_state_machine(
+            stateMachineArn=state_machine_arn, definition=definition_r3_str
+        )
+        sfn_snapshot.match("update_resp_3", update_resp_3)
+
+        with pytest.raises(Exception) as conflict_exception:
+            aws_client.stepfunctions.publish_state_machine_version(
+                stateMachineArn=state_machine_arn, revisionId=revision_id_r2
+            )
+        sfn_snapshot.match("conflict_exception", conflict_exception.value)
+
+    @markers.snapshot.skip_snapshot_verify(
+        # TODO: status and output update flaky on aws when listing versions.
```

Comment:
can you explain the flakiness here a bit more in detail? how does this show?

Also asking because skip_snapshot_verify won't help with flakiness on AWS

Reply:
Good point thank you, I decided to resolve this now. Aws seem to had some inconsistencies in listing the status of an execution between their events history and list executions outputs. I added `await_execution_lists_terminated` that waits for these two output to align.